tffm/models.py

```diff
@@ -27,19 +31,18 @@ def _preprocess_sample_weights(self, sample_weight, pos_class_weight, used_y):
         used_w = np.ones_like(used_y)
         if sample_weight is None and pos_class_weight is None:
             return used_w
-
         if type(pos_class_weight) == float:
             used_w[used_y > 0] = pos_class_weight
         elif sample_weight == "balanced":
             pos_rate = np.mean(used_y > 0)
             neg_rate = 1 - pos_rate
-            used_w[used_y > 0] = 1.0 / pos_rate / 2
-            used_w[used_y < 0] = 1.0 / neg_rate / 2
+            used_w[used_y > 0] = neg_rate / pos_rate
+            used_w[used_y < 0] = 1.0
```

Comment:
These SHOULD be equivalent, but they weren't. Since they cause the loss function to differ by a scaling factor, perhaps the effective learning rate changes or something like that. Anyways, the one I settled on ended up giving more reasonable performance.